ledger/acctupdates.go

```diff
@@ -374,7 +387,13 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 			rnd = currentDbRound + basics.Round(currentDeltaLen)
 		}
 
-		// OTHER LOOKUPS USE "base" caches here.
+		// check the baseBoxes cache
+		if pbd, has := au.baseBoxes.read(key); has {
+			// we don't technically need this, since it's already in the baseBoxes, however, writing this over
+			// would ensure that we promote this field.
+			au.baseBoxes.writePending(pbd, key)
```

Comment:
question: even if we read it in cache, we also rerank the key to be the latest touched, right?

Reply:
Yes, that's the purpose of the code you highlighted. It's a little more complicated though because the writes are first buffered than flushed later. The higher level goal is the same.